code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -510,31 +510,31 @@ GLOBAL_LIST_INIT(durathread_recipes, list ( \
 	force = 0
 	throwforce = 0
 	merge_type = /obj/item/stack/sheet/cotton
-	var/pull_effort = 1 SECONDS
-	var/loom_result = /obj/item/stack/sheet/cloth
 	grind_results = list(/datum/reagent/cellulose = 20)
+	var/loom_result = /obj/item/stack/sheet/cloth
+	var/loom_time = 1 SECONDS
 
 /obj/item/stack/sheet/cotton/Initialize(mapload)
 	. = ..()
-	AddComponent(/datum/component/loomable, resulting_item = loom_result, loom_time = pull_effort)
+	AddElement(/datum/element/loomable, resulting_atom = src.loom_result, loom_time = src.loom_time)
```

Comment:
do we _really_ need `src` here? it's really redundant- the only reason why we use `src` in AddElement itself (and not the args) is because of this in the style guide: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar

this should just simply be:
```suggestion
	AddElement(/datum/element/loomable, resulting_atom = loom_result, loom_time = loom_time)
```

I know `loom_time = loom_time` is silly but it's obvious in this context that the left-hand side is the argument name and the right-hand side is the value that we read off the sheet-stack. it's not obvious back in the actual element code, but i think it makes enough sense here to not have to do `src` memes like this (shadow jutsu is to change the name of the sheet-stack's var and dmdoc it)